### PR TITLE
[DDW-702] Fix screen loading logic

### PR DESCRIPTION
**Pull request by daniloprates**
This PR fixes screen flashes during Daedalus loading.

https://iohk.myjetbrains.com/youtrack/_persistent/bug.mp4?file=88-1730&c=false&updated=1560454658606

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1562017384187000)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
Reviewing this now...

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 1]
```diff
@@ -153,16 +154,19 @@ export default class NetworkStatusStore extends Store {
     );
   };
 
-  @action async _restartNode() {
+  _restartNode = async () => {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @daniloprates why did we need to do this change? Can't this method remain an action?

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 2]
```diff
@@ -488,6 +492,12 @@ export default class NetworkStatusStore extends Store {
             (isLocalBlockHeightSyncing || isNetworkBlockHeightSyncing);
         });
 
+        if (this.isNodeRestarting && this.isNodeSyncing) {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @daniloprates can you please clarify this part? 
Wouldn't it be better to shift this logic which sets `this.isNodeRestarting = false;` under `_handleCardanoNodeStateChange` method?

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 1 above)
Comment (daniloprates) [reply source/renderer/app/stores/NetworkStatusStore.js]: The main purpose of this function is to restart node, which doesn't require it to be an action, as we call `await restartCardanoNodeChannel.send();`. 

I'm not entirely sure why, but as an action it loses the Class instance and `this` becomes a global object, so `this.isNodeRestarting = true` doesn't work. 

With these changes, the main function is not an action, as it doesn't need to be, and `this.isNodeRestarting = true;` is wrapped into a `runInAction` method.

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: OK thanks for the answer @daniloprates

File: source/renderer/app/stores/NetworkStatusStore.js (hunk 2 above)
Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: @daniloprates I still think this is worth improving 🙏

Comment (daniloprates) [reply source/renderer/app/stores/NetworkStatusStore.js]: The reason it needs to be there is that, after Node is Running, there's a gap before `hasStartedReceivingBlocks` is `true`, which is when the flickering happens.

The way it was done, not only awaits for Node to be Running, but also for the first block to be received.

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: @daniloprates are you sure? `isConnected` requires for the `isNodeSyncing` to be `true` which implies that `hasStartedReceivingBlocks` is `true`?

Comment (daniloprates) [reply source/renderer/app/stores/NetworkStatusStore.js]: Investigating

Comment (daniloprates) [reply source/renderer/app/stores/NetworkStatusStore.js]: Found the issue. `_requestCardanoStatus` was overriding the current status, wrongly setting `isNodeSyncing` to `true`. Simplified the code, now.

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: Tanks @daniloprates - will review this first thing tomorrow morning!

File: source/renderer/app/stores/NetworkStatusStore.js [hunk 3]
```diff
@@ -233,8 +233,9 @@ export default class NetworkStatusStore extends Store {
       if (status)
         runInAction('assigning node status', () => {
           const { cardanoNodeID } = status;
+          const { isNodeSyncing } = this;
           this.cardanoNodeID = cardanoNodeID;
-          Object.assign(this, status);
+          Object.assign(this, status, { isNodeSyncing });
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NetworkStatusStore.js]: @daniloprates I think this is not a proper fix. The problem appears to be that we do not update cardano status if we are not connected: https://github.com/input-output-hk/daedalus/blob/40ea71280e85f43d6565667a019513ee40c0a810/source/renderer/app/stores/NetworkStatusStore.js#L193
Let's chat about this once you start today...

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: We need to be careful not to break something else. This caching was introduced by @DominikGuzei in the scope of this commit: https://github.com/input-output-hk/daedalus/commit/60d292f3f6e37e5acad3d619b2b10683e908a275

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: Which was a part of this PR: https://github.com/input-output-hk/daedalus/pull/1178

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: Perhaps we should cache this status only in test mode? @DominikGuzei @daniloprates

Comment (DominikGuzei) [reply source/renderer/app/stores/NetworkStatusStore.js]: Caching only in test mode would be a quick solution 👍  I don't think it brings much value for the user, only for app reloads which probably happen not that often

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: @DominikGuzei let's discuss this today on our weekly call 🙏

Comment (nikolaglumac) [reply source/renderer/app/stores/NetworkStatusStore.js]: Done! @DominikGuzei @daniloprates

**Comment by nikolaglumac** [comment]
This one requires no QA. @daniloprates please review and test it once again and then we can merge it.

**Comment by nikolaglumac** [comment]
@DominikGuzei please review this one too 🙏

**Pull request merged**